### PR TITLE
Rgb20 iface pair global name assignment name transistion name

### PR DESCRIPTION
**Pull request by cymqqqq**

### CODE REVIEW

File: src/interface/iimpl.rs [hunk 1]
```diff
@@ -292,16 +292,33 @@ impl IfacePair {
...
     pub fn iface_id(&self) -> IfaceId { self.iface.iface_id() }
+
     pub fn impl_id(&self) -> ImplId { self.iimpl.impl_id() }
+
     pub fn global_type(&self, name: &FieldName) -> Option<GlobalStateType> {
         self.iimpl.global_type(name)
     }
+
+    pub fn global_name(&self, global_type: &GlobalStateType) -> Option<&FieldName> {
+        self.iimpl.global_name(global_type.to_owned())
```
Comment (dr-orlovsky) [inline src/interface/iimpl.rs:303]: This type is `Copy`, thus there is no need for `to_owned()`. Just change the argument to `global_type: GlobalStateType`

File: src/interface/iimpl.rs [hunk 2]
```diff
@@ -292,16 +292,33 @@ impl IfacePair {
...
+    }
+
     pub fn assignments_type(&self, name: &FieldName) -> Option<AssignmentType> {
         self.iimpl.assignments_type(name)
     }
+
+    pub fn assignment_name(&self, assignment_type: &AssignmentType) -> Option<&FieldName> {
```
Comment (dr-orlovsky) [inline src/interface/iimpl.rs:310]: ibid

File: src/interface/iimpl.rs [hunk 3]
```diff
@@ -292,16 +292,33 @@ impl IfacePair {
...
+        self.iimpl.assignment_name(assignment_type.to_owned())
+    }
+
     pub fn transition_type(&self, name: &TypeName) -> Option<TransitionType> {
         self.iimpl.transition_type(name)
     }
+
+    pub fn transition_name(&self, transition_type: &TransitionType) -> Option<&TypeName> {
```
Comment (dr-orlovsky) [inline src/interface/iimpl.rs:318]: ibid

**Review by dr-orlovsky — COMMENTED** [review]
LGTM except lints in the arguments

File: src/interface/iimpl.rs (hunk 1 above)
Comment (cymqqqq) [reply src/interface/iimpl.rs]: done

**Pull request merged**